server/controllers/controller_helpers/metrics.js

```diff
@@ -0,0 +1,115 @@
+import _ from 'lodash';
+
+import responsesService from '../../db/services/responses';
+import campaignsService from '../../db/services/campaigns';
+
+function splitResponses(responseString) {
+  return responseString.split(',');
+}
+
+export function mapCallIds(callObjArray) {
+  return callObjArray.map(call => call.attributes.id);
+}
+
+export function checkCampaignNotDraft(campaignId, res) {
+  return campaignsService.getCampaignById({ id: campaignId })
+    .then((campaign) => {
+      if (campaign) {
+        const { status } = campaign.attributes;
+        if (status === 'draft') {
+          throw new Error('Campaign metrics not available for campaigns still in draft.');
+        }
+        return true;
+      }
+      return res.status(500).json({ message: 'problem with db query for campaign by id' });
+    })
+    .catch((err) => {
+      const errorMessage = `could not find campaigin by id in campaign status check (metrics): ${err}`;
+      return res.status(404).json({ message: errorMessage });
+    });
+}
+
+function createStatusMap(metricsObj) {
```

Comment:
This is a design choice, but I'm wondering if it makes sense to abstract out the metricsObj rather than passing it into each of these helpers? Then add them to the metricsObj in countStatusAndOutcomeFrequency and handleQuestionResponses?

Reply:
Yeah, I thought it was a bit more readable this way, instead of having assignments, the function invocations read as what they're doing. But I def see what you mean. I think it's just preference, do you see a performance benefit either way?

Reply:
I think it's written properly, but I suppose it's possible that unrelated part of the metricsObj could get updated by accident? Each function is only updating part of it, but we're passing the whole thing in, you know?

Reply:
I think in the interest of time we can leave it as is.

Reply:
Ok, coo beans.

Reply:
*cool